src/testnet/node.rs

```diff
@@ -0,0 +1,385 @@
+            res.first().unwrap().0.as_ref().unwrap().clone() // todo(ludo): yikes
+        };
+
+        self.chain_tip = Some(new_chain_tip);
+
+        // Unset the `bootstraping_chain` flag.
+        if self.bootstraping_chain {
+            self.bootstraping_chain = false;
+        }
+    }
+
+    /// Returns the Stacks address of the node
+    pub fn get_address(&self) -> StacksAddress {
+        self.keychain.get_address()
+    }
+
+    /// Constructs and returns a LeaderKeyRegisterOp out of the provided params
+    fn generate_leader_key_register_op(&mut self, vrf_public_key: VRFPublicKey, consensus_hash: ConsensusHash) -> BlockstackOperationType {
+
+        BlockstackOperationType::LeaderKeyRegister(LeaderKeyRegisterOp {
+            public_key: vrf_public_key,
+            memo: vec![],
+            address: self.keychain.get_address(),
+            consensus_hash,
+
+            // Props that will be set by the burnchain simulator
+            vtxindex: 0,
+            txid: Txid([0u8; 32]),
+            block_height: 0,
+            burn_header_hash: BurnchainHeaderHash([0u8; 32]),
+        })
+    }
+
+    /// Constructs and returns a LeaderBlockCommitOp out of the provided params
+    fn generate_block_commit_op(&mut self, 
+                                block_header_hash: BlockHeaderHash,
+                                burn_fee: u64, 
+                                key: &RegisteredKey,
+                                parent_block: &SortitionedBlock,
+                                vrf_seed: VRFSeed) -> BlockstackOperationType {
+        
+        let (parent_block_ptr, parent_vtxindex) = match self.bootstraping_chain {
+            true => (0, 0), // Expected references when mocking the initial sortition
+            false => (parent_block.block_height as u32, parent_block.op_vtxindex as u16)
+        };
+
+        BlockstackOperationType::LeaderBlockCommit(LeaderBlockCommitOp {
+            block_header_hash,
+            burn_fee,
+            input: self.keychain.get_burnchain_signer(),
+            key_block_ptr: key.block_height as u32,
+            key_vtxindex: key.op_vtxindex as u16,
+            memo: vec![],
+            new_seed: vrf_seed,
+            parent_block_ptr,
+            parent_vtxindex,
+
+            // Props that will be set by the burnchain simulator
+            vtxindex: 0,
+            txid: Txid([0u8; 32]),
+            block_height: 0,
+            burn_header_hash: BurnchainHeaderHash([0u8; 32]),
+        })
+    }
+
+    // Constructs a coinbase transaction
+    fn generate_coinbase_tx(&mut self) -> StacksTransaction {
+        let mut tx_auth = self.keychain.get_transaction_auth().unwrap();
+        tx_auth.set_origin_nonce(self.nonce);
+
+        let mut tx = StacksTransaction::new(
+            TransactionVersion::Testnet, 
+            tx_auth, 
+            TransactionPayload::Coinbase(CoinbasePayload([0u8; 32])));
+        tx.chain_id = 0x80000000;
```

Comment:
I'd recommend using a constant variable for `0x80000000` to indicate that this is the testnet chain :)

Reply:
Addressed with https://github.com/blockstack/blockstack-core/pull/1202/commits/33cec8814fa537a9e4d56184dc56b18d8bdf60be